Sources/FTAPIKit/Coding.swift

```diff
@@ -4,23 +4,36 @@ import Foundation
 import FoundationNetworking
 #endif
 
+/// `Encoding` is not only represents Swift encoders, but also provides network specific feature, such as
+/// configuring the request with correct headers.
+///
+/// - Note: A reference implementation is provided in the form of `JSONEncoding`
 public protocol Encoding {
+
+    /// Encodes the argument
     func encode<T: Encodable>(_ object: T) throws -> Data
+
+    /// Sets correct header to the request, such as `Content-Type`
     func configure(request: inout URLRequest) throws
 }
 
+/// `Decoding` encapsulates Swift decoders.
 public protocol Decoding {
     func decode<T: Decodable>(data: Data) throws -> T
 }
 
+/// Reference implementation of `Encoding` using JSON `Foundation.JSONEncoder` under the hood.
 public struct JSONEncoding: Encoding {
     private let encoder: JSONEncoder
 
     public init(encoder: JSONEncoder = .init()) {
         self.encoder = encoder
     }
 
-    public init(configure: (JSONEncoder) -> Void) {
+    /// This initializer is a syntax sugar that provides the the possibility to configure the `JSONEncoder` in
+    /// a compact manner.
+    /// - Parameter encoder: Encoder which will be used by the instance
```

Comment:
The parameter is documenting the closure parameter, not the function parameter.

```suggestion
    /// Creates new encoder with ability to configure `JSONEncoder` in a compact manner
    /// using a closure.
    ///
    /// - Parameter configure: Function with custom configuration of the encoder
```